app/main.py

```diff
@@ -1 +1,48 @@
-#  write your code here
+class Animal:
+
+    def __init__(
+            self, name: str,
+            appetite: int,
+            is_hungry: bool = True
+    ) -> None:
+        self. name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if not self.is_hungry:
+            return 0
+
+        print(f"Eating {self.appetite} food points...")
+        self.is_hungry = not self.is_hungry
+        return self.appetite
+
+
+class Cat(Animal):
+
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, appetite=3, is_hungry=is_hungry)
+
+    @staticmethod
+    def catch_mouse() -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, appetite=7, is_hungry=is_hungry)
+
+    @staticmethod
+    def bring_slippers() -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(hungry_animals: list) -> int:
+    return sum(
+        feed_all_animals.feed()
+        for feed_all_animals in hungry_animals
```

Comment:
You should consider to rename variables
Method feed_animals gets not only hungry_animals. Thats why you have `if not self.is_hungry:` check in feed animals
Also in the loop you iterate by _animal_, so `feed_all_animals` is a bad name here

Reply:
I see, I can get all animals and feed hungry animals